### PR TITLE
[#681] UG: Update UG in accordance to change in Toolbar

### LINKED ISSUE #681: UG: outdated Tool Bar section
As title.

missing `per commit` `Granularity`:
image

outdated `reverse`, `group`, `sort by`:
image

Should make consistant with below:
image

### LINKED ISSUE #756: UG: Provide a more appropriate example of cli command
In the User Guide, there is an example of a cli command:
Screen Shot 2019-06-19 at 10 00 09 PM

That command will lead to this page where there are no commits in between the since date
and until date:
Screen Shot 2019-06-19 at 10 05 56 PM

Suggestion: Let's update the since date and until date of that cli command example that can capture commits to provide a more realistic command.

### PR DESCRIPTION
**Pull request by jylee-git**
```
The recent changes in the toolbar component of the report view is not 
reflected in the user guide, especially the "group by" function.

When users want to group by a specific category, they may not 
understand as the user guide relating to the toolbar is outdated.

Let's update the user guide to reflect the changes made to 
the Toolbar. Let's also take this chance to rephrase certain 
descriptions that are outdated or misleading.
```
Resolves #681 
Resolves #756

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -98,17 +98,20 @@ The `Tool Bar` at the top provides a set of configuration options that control t
...
+  * `None` : results will not be grouped in any particular way.
+  * `Repo/Branch` : results will be grouped by repositories and its' associating branches.
+  * `Author` : results will be  grouped by the name of the author. Contributions made to multiple repositories by a particular author will be grouped under the author.
 * `Sort by` : sorting criteria for the Chart Panel
   * `Total Contribution` : the amount of lines, written by the author, in the repository.
   * `Variance` : the [variance](https://en.wikipedia.org/wiki/Variance) of the number of lines that the author has contributed to the repository daily, sorted from low to high.
   * `Author Name` : the author's display name.
   * `Repo/Branch Name` : see note [1] below.
 * `Granularity` : the period of time for which commits are aggregated in the Ramp Chart.
+    * `Commit`: each commit made is represented as one ramp
```
Comment (fzdy1914) [inline docs/UserGuide.md]: ~represented~ -> `shown`

File: docs/UserGuide.md [hunk 2]
```diff
@@ -98,17 +98,20 @@ The `Tool Bar` at the top provides a set of configuration options that control t
   * Multiple keywords/terms can be used, separated by spaces.
   * Entries that contain _any_ (not necessarily _all_) of the search terms will be displayed.
   * The keywords used to filter author and repository are case-insensitive.
+* `Group by` : denotes whether to group the captured results by
+  * `None` : results will not be grouped in any particular way.
+  * `Repo/Branch` : results will be grouped by repositories and its' associating branches.
+  * `Author` : results will be  grouped by the name of the author. Contributions made to multiple repositories by a particular author will be grouped under the author.
 * `Sort by` : sorting criteria for the Chart Panel
```
Comment (fzdy1914) [inline docs/UserGuide.md]: ~Sort by~ -> `sort group by` and `sort  within group by`

File: docs/UserGuide.md [hunk 3]
```diff
@@ -98,17 +98,30 @@ The `Tool Bar` at the top provides a set of configuration options that control t
   * Multiple keywords/terms can be used, separated by spaces.
   * Entries that contain _any_ (not necessarily _all_) of the search terms will be displayed.
   * The keywords used to filter author and repository are case-insensitive.
-* `Sort by` : sorting criteria for the Chart Panel
-  * `Total Contribution` : the amount of lines, written by the author, in the repository.
-  * `Variance` : the [variance](https://en.wikipedia.org/wiki/Variance) of the number of lines that the author has contributed to the repository daily, sorted from low to high.
-  * `Author Name` : the author's display name.
-  * `Repo/Branch Name` : see note [1] below.
+
```
Comment (fzdy1914) [inline docs/UserGuide.md]: Unnecessary blank line. Below lines are same.

File: docs/UserGuide.md [hunk 4]
```diff
@@ -98,17 +98,30 @@ The `Tool Bar` at the top provides a set of configuration options that control t
...
+* `Sort within groups by`: sorting criteria within a group
+  * `Name` : sorted by GitHub ID in alphabetical order.
+  * `Repo/branch` : sorted in alphabetical order by the name of the repo, followed by name of the branch. See note [1] below.
+  * `Contribution` : sorted by individual contributions in the order of number of lines added
+  * `Variance` : the [variance](https://en.wikipedia.org/wiki/Variance) of the number of lines that the group has contributed to the repository daily.
+  
 * `Granularity` : the period of time for which commits are aggregated in the Ramp Chart.
+    * `Commit`: each commit made is shown as one ramp
     * `Day`: commits within a day (commits made within 00:00 to 23:59 in **GMT+8**) are shown as one ramp
```
Comment (fzdy1914) [inline docs/UserGuide.md]: I found `**GMT+8**` is not correct, too. We can customize the timezone now. Can you kindly update them as well?

**Comment by jylee-git** [comment]
Does RepoSense report support browsers other than chrome? From my side Microsoft Edge isn't compatible with RepoSense (can't load code view). 

If only chrome is supported we can update it in the UG as well.

**Comment by fzdy1914** [comment]
> Does RepoSense report support browsers other than chrome? From my side Microsoft Edge isn't compatible with RepoSense (can't load code view).
> 
> If only chrome is supported we can update it in the UG by adding chrome in the prerequisites list.

No, we should support all major browsers. You may raise an issue about this.

**Comment by jylee-git** [comment]
> > Does RepoSense report support browsers other than chrome? From my side Microsoft Edge isn't compatible with RepoSense (can't load code view).
> > If only chrome is supported we can update it in the UG by adding chrome in the prerequisites list.
> 
> No, we should support all major browsers. You may raise an issue about this.

Noted. Will raise an issue

File: docs/UserGuide.md [hunk 5]
```diff
@@ -284,7 +292,7 @@ The directory used with the `--config` parameter should contain a `repo-config.c
 `repo-config.csv` file contains repo-level config data as follows:
 
 * First row: column headings, ignored by RepoSense
-* Second row: repo data
+* Second row onwards: repo data
```
Comment (eugenepeh) [inline docs/UserGuide.md]: each row represents a repository's configuration

File: docs/UserGuide.md [hunk 6]
```diff
@@ -105,17 +105,25 @@ The `Tool Bar` at the top provides a set of configuration options that control t
   * Multiple keywords/terms can be used, separated by spaces.
   * Entries that contain _any_ (not necessarily _all_) of the search terms will be displayed.
   * The keywords used to filter author and repository are case-insensitive.
-* `Sort by` : sorting criteria for the Chart Panel
-  * `Total Contribution` : the amount of lines, written by the author, in the repository.
-  * `Variance` : the [variance](https://en.wikipedia.org/wiki/Variance) of the number of lines that the author has contributed to the repository daily, sorted from low to high.
-  * `Author Name` : the author's display name.
-  * `Repo/Branch Name` : see note [1] below.
+* `Group by` : denotes whether to group the captured results by
```
Comment (eugenepeh) [inline docs/UserGuide.md]: word it accordingly like the sort by or vice versa.

grouping criteria for the rows of results

File: docs/UserGuide.md [hunk 7]
```diff
@@ -105,17 +105,25 @@ The `Tool Bar` at the top provides a set of configuration options that control t
...
+  * `None` : results will not be grouped in any particular way.
+  * `Repo/Branch` : results will be grouped by repositories and its' associating branches.
+  * `Author` : results will be  grouped by the name of the author. Contributions made to multiple repositories by a particular author will be grouped under the author.  
+* `Sort groups by`: sorting criteria for the main group
+  * `Name` : groups will be sorted by GitHub ID in alphabetical order.
+  * `Repo/branch` : groups will be sorted in alphabetical order by the name of the repo, followed by name of the branch. See note [1] below.
+  * `Contribution` : groups will be sorted by the total combined contributions within a group, in the order of number of lines added
```
Comment (eugenepeh) [inline docs/UserGuide.md]: groups will be sorted by the ~total~ combined contributions of each group, based on the number of lines written by the authors

File: docs/UserGuide.md [hunk 8]
```diff
@@ -105,17 +105,25 @@ The `Tool Bar` at the top provides a set of configuration options that control t
...
+  * `Variance` : the [variance](https://en.wikipedia.org/wiki/Variance) of the number of lines that the group has contributed to the repository daily.
+* `Sort within groups by`: sorting criteria within a group
```
Comment (eugenepeh) [inline docs/UserGuide.md]: each group

**Comment by jylee-git** [comment]
Should I be making this PR to the release branch instead of the master branch?

File: docs/UserGuide.md [hunk 9]
```diff
@@ -105,17 +105,25 @@ The `Tool Bar` at the top provides a set of configuration options that control t
...
   * The keywords used to filter author and repository are case-insensitive.
-* `Sort by` : sorting criteria for the Chart Panel
-  * `Total Contribution` : the amount of lines, written by the author, in the repository.
-  * `Variance` : the [variance](https://en.wikipedia.org/wiki/Variance) of the number of lines that the author has contributed to the repository daily, sorted from low to high.
-  * `Author Name` : the author's display name.
-  * `Repo/Branch Name` : see note [1] below.
+* `Group by` : grouping criteria for the rows of results
+  * `None` : results will not be grouped in any particular way.
+  * `Repo/Branch` : results will be grouped by repositories and its' associating branches.
+  * `Author` : results will be  grouped by the name of the author. Contributions made to multiple repositories by a particular author will be grouped under the author.
```
Comment (eugenepeh) [inline docs/UserGuide.md]: theres a double space in the sentence

File: docs/UserGuide.md [hunk 10]
```diff
@@ -105,17 +105,25 @@ The `Tool Bar` at the top provides a set of configuration options that control t
...
+  * `Name` : sorted by GitHub ID in alphabetical order.
+  * `Repo/branch` : sorted in alphabetical order by the name of the repo, followed by name of the branch. See note [1] below.
+  * `Contribution` : sorted by individual contributions in the order of number of lines added
+  * `Variance` : the [variance](https://en.wikipedia.org/wiki/Variance) of the number of lines that the group has contributed to the repository daily.
 * `Granularity` : the period of time for which commits are aggregated in the Ramp Chart.
-    * `Day`: commits within a day (commits made within 00:00 to 23:59 in **GMT+8**) are shown as one ramp
-    * `Week`:  commits within a week are shown as one ramp
+    * `Commit`: each commit made is shown as one ramp
+    * `Day`: commits within a day (commits made within 00:00 to 23:59) are shown as one ramp
+    * `Week`:  commits within a week are shown as one ramp
```
Comment (eugenepeh) [inline docs/UserGuide.md]: same for here

File: docs/UserGuide.md [hunk 11]
```diff
@@ -105,17 +105,25 @@ The `Tool Bar` at the top provides a set of configuration options that control t
...
+* `Sort groups by`: sorting criteria for the main group
+  * `Name` : groups will be sorted by GitHub ID in alphabetical order.
+  * `Repo/branch` : groups will be sorted in alphabetical order by the name of the repo, followed by name of the branch. See note [1] below.
+  * `Contribution` : groups will be sorted by the combined contributions within a group, in the order of number of lines added
+  * `Variance` : the [variance](https://en.wikipedia.org/wiki/Variance) of the number of lines that the group has contributed to the repository daily.
+* `Sort within groups by`: sorting criteria within each group
+  * `Name` : sorted by GitHub ID in alphabetical order.
+  * `Repo/branch` : sorted in alphabetical order by the name of the repo, followed by name of the branch. See note [1] below.
+  * `Contribution` : sorted by individual contributions in the order of number of lines added
+  * `Variance` : the [variance](https://en.wikipedia.org/wiki/Variance) of the number of lines that the group has contributed to the repository daily.
```
Comment (eugenepeh) [inline docs/UserGuide.md]: It seems like the description for `variance` follows a very different style from the other 3. Also, the description is the same as the  `Sort groups by`.

The description for the fields under the `Sort within groups by` also follows a very different style from the other descriptions.

Try to maintain the same style, either blends into the existing style or standardize all the description. Having too many different styles in a same paragraph can easily confuse the users.

Comment (jylee-git) [reply docs/UserGuide.md]: I re-wrote the description for variance part as I believe the previous version isn't well defined. Do let me know if the description needs to be improved further.

File: docs/UserGuide.md [hunk 12]
```diff
@@ -113,12 +113,12 @@ The `Tool Bar` at the top provides a set of configuration options that control t
...
 * `Sort within groups by`: sorting criteria within each group
-  * `Name` : sorted by GitHub ID in alphabetical order.
-  * `Repo/branch` : sorted in alphabetical order by the name of the repo, followed by name of the branch. See note [1] below.
-  * `Contribution` : sorted by individual contributions in the order of number of lines added
-  * `Variance` : the [variance](https://en.wikipedia.org/wiki/Variance) of the number of lines that the group has contributed to the repository daily.
+  * `Name` : each sub-group will be sorted by GitHub ID in alphabetical order.
+  * `Repo/branch` : each sub-group will be sorted in alphabetical order by the name of the repo, followed by name of the branch. See note [1] below.
+  * `Contribution` : each sub-group will be sorted by individual contributions in the order of number of lines added
+  * `Variance` : each sub-group will be sorted by the average of the squared differences from the average number of lines of code contributed per day among all authors involved. Detailed definition of variance is located [here](https://en.wikipedia.org/wiki/Variance).
```
Comment (eugenepeh) [inline docs/UserGuide.md]: this should be base on individual

File: docs/UserGuide.md [hunk 13]
```diff
@@ -269,10 +269,10 @@ Example:`--view` or `-v`
   Example:`--timezone UTC+08` or `-t UTC-1030`
 
 Here's an example of a command using all parameters:<br>
-`java -jar RepoSense.jar --repo https://github.com/reposense/RepoSense.git --output ./report_folder --since 21/10/2017 --until 21/11/2017 --formats java adoc js --view --ignore-standalone-config --timezone UTC+08`
+`java -jar RepoSense.jar --repo https://github.com/se-edu/addressbook-level4.git --output ./report_folder --since 1/1/2017 --until 1/1/2018 --formats java adoc xml --view --ignore-standalone-config --timezone UTC+08`
```
Comment (eugenepeh) [inline docs/UserGuide.md]: why this change?

Comment (jylee-git) [reply docs/UserGuide.md]: The previous example leads to an empty report as no commit infos were available during that period given the parameters. #756

Comment (eugenepeh) [reply docs/UserGuide.md]: I see. That is fine but avoid using a repository from different organization in our documentation unless its for acknowledgement. Also, while the date range is fine, try to use dates that allows user to easily tells apart the day and the month. E.g. `31/10`.

Comment (jylee-git) [reply docs/UserGuide.md]: Ok. We use back RepoSense in this case?

Comment (eugenepeh) [reply docs/UserGuide.md]: Yes

Comment (damithc) [reply docs/UserGuide.md]: > That is fine but avoid using a repository from different organization in our documentation unless its for acknowledgement.

I suggested using se-edu because using `RepoSense` multiple times in the command can be confusing to the reader as it is also the name of the tool.

Comment (damithc) [reply docs/UserGuide.md]: But I'm OK to keep using RepoSense repo too.

**Comment by jylee-git** [comment]
CI checks done @eugenepeh
